validator/bundledschemalocator.go

```diff
@@ -0,0 +1,52 @@
+// Copyright Axis Communications AB.
+//
+// For a full list of individual contributors, please see the commit history.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path/filepath"
+)
+
+//go:generate go run ./../internal/cmd/dirsync ../protocol/schemas/ ./schemas
+
+//go:embed schemas
+var schemas embed.FS
+
+// BundledSchemaLocator locates schemas from those built into the binary via this package.
+type BundledSchemaLocator struct{}
```

Comment:
Do you plan using the struct for something, or why can't GetSchema be a regular function?
I thought it was for testability, but I see you don't use it that way (yet?).

Reply:
SchemaLocator interface
https://github.com/eiffel-community/eiffelevents-sdk-go/pull/71/files#diff-6e53047132b79f5507f0dc1222bddb1da0a5a94b6b9126144a4f01a34d36f458R36

Reply:
Indeed. The SchemaLocator interface exists so an SDK client can provide additional means of supplying schemas to the validator, e.g. if you have your own events (unusual, but totally fine) or if you don't want to update the SDK dependency and redeploy just to be able to get access to new schemas that are included via `//go:embed`.